logging/jboss/src/test/java/io/quarkus/ts/logging/jboss/AccessLoggingIT.java

```diff
@@ -0,0 +1,162 @@
+
+    @Test
+    public void logRotationTest() throws IOException {
+        // use zip compression for archived logs
+        setPropertyTo("quarkus.log.handler.file.access-log.rotation.file-suffix", "yyyy-MM-dd.zip");
+
+        // one request should do about 400-500B
+        // We have configured max 5KB for one log file, and max two archive log files
+        // 50 requests should be enough to fill this all
+        // sending more requests to check that log archives are actually discarded when they go over limit
+        sendRequests(80);
+
+        List<String> logFileNames = getLogFileNames();
+        // there should be 3 files - one active log and two archives/backups
+        assertEquals(3, logFileNames.size(), "There should be 3 log file - one active and two backups");
+
+        // check main log file size, we have max 5KB configured
+        assertTrue(Files.size(accessLogPath()) <= 5120, "Main log size should be max 5KB");
+
+        // check that archive logs are valid zip files
+        for (String filename : logFileNames) {
+            // we don't check the main log file, that one is not archived nor compressed
+            if (filename.equals(ACCESS_LOG_FILENAME)) {
+                continue;
+            }
+
+            assertTrue(filename.endsWith(".zip"), "Archived logs should have suffix \".zip\"");
+            assertTrue(isValidZipFile(ACCESS_LOG_DIR.resolve(filename)), "Log file " + filename + " is not a valid zip");
+        }
+    }
+
+    @Test
+    public void gzipCompressionTest() throws IOException {
+        // use gzip compression for archived logs
+        setPropertyTo("quarkus.log.handler.file.access-log.rotation.file-suffix", "yyyy-MM-dd.gz");
+
+        sendRequests(50);
+
+        List<String> logFileNames = getLogFileNames();
+        // there should be 3 files - one active log and two archives/backups
+        assertEquals(3, logFileNames.size(), "There should be 3 log file - one active and two backups");
+
+        // check that logs are valid .gz files
+        for (String filename : logFileNames) {
+            // we don't check the main log file, that one is not archived nor compressed
+            if (filename.equals(ACCESS_LOG_FILENAME)) {
+                continue;
+            }
+
+            assertTrue(filename.endsWith(".gz"), "Archived logs should have suffix \".gz\"");
+            assertTrue(isValidGzipFile(ACCESS_LOG_DIR.resolve(filename)), "Log file " + filename + " is not a valid zip");
+        }
+    }
+
+    private void setPropertyTo(String property, String value) {
+        app.stop();
+        app.withProperty(property, value);
+        app.start();
+    }
+
+    private boolean isValidGzipFile(Path file) throws IOException {
+        try (InputStream is = new FileInputStream(file.toFile())) {
+            byte[] signature = new byte[2];
+            int nread = is.read(signature);
+            return nread == 2 && signature[0] == (byte) 0x1f && signature[1] == (byte) 0x8b;
```

Comment:
A link to specification would be appreciated

Reply:
added